src/cli/serve/serve.js

```diff
@@ -209,31 +222,32 @@ export default function (program) {
     }
 
     const unknownOptions = this.getUnknownOptions();
-    await bootstrap({
-      configs: [].concat(opts.config || []),
-      cliArgs: {
-        dev: !!opts.dev,
-        envName: unknownOptions.env ? unknownOptions.env.name : undefined,
-        // no longer supported
-        quiet: !!opts.quiet,
-        silent: !!opts.silent,
-        watch: !!opts.watch,
-        runExamples: !!opts.runExamples,
-        // We want to run without base path when the `--run-examples` flag is given so that we can use local
-        // links in other documentation sources, like "View this tutorial [here](http://localhost:5601/app/tutorial/xyz)".
-        // We can tell users they only have to run with `yarn start --run-examples` to get those
-        // local links to work.  Similar to what we do for "View in Console" links in our
-        // elastic.co links.
-        basePath: opts.runExamples ? false : !!opts.basePath,
-        optimize: !!opts.optimize,
-        disableOptimizer: !opts.optimizer,
-        oss: !!opts.oss,
-        cache: !!opts.cache,
-        dist: !!opts.dist,
-      },
-      features: {
-        isCliDevModeSupported: DEV_MODE_SUPPORTED,
-      },
+    const configs = [].concat(opts.config || []);
+    const cliArgs = {
+      dev: !!opts.dev,
+      envName: unknownOptions.env ? unknownOptions.env.name : undefined,
+      // no longer supported
+      quiet: !!opts.quiet,
+      silent: !!opts.silent,
+      watch: !!opts.watch,
+      runExamples: !!opts.runExamples,
+      // We want to run without base path when the `--run-examples` flag is given so that we can use local
+      // links in other documentation sources, like "View this tutorial [here](http://localhost:5601/app/tutorial/xyz)".
+      // We can tell users they only have to run with `yarn start --run-examples` to get those
+      // local links to work.  Similar to what we do for "View in Console" links in our
+      // elastic.co links.
+      basePath: opts.runExamples ? false : !!opts.basePath,
+      optimize: !!opts.optimize,
+      disableOptimizer: !opts.optimizer,
+      oss: !!opts.oss,
+      cache: !!opts.cache,
+      dist: !!opts.dist,
+    };
+    const bootstrapScript = getBootstrapScript(cliArgs.dev);
+
+    await bootstrapScript({
```

Comment:
@spalger What is the main reason we want `dev_cli` controlling the Kibana server creation? It always confuses me when I tried to debug the Kibana server with Chrome https://github.com/elastic/kibana/blob/77e56259106a259fa2bb69497aaf08aa3224cacf/packages/kbn-cli-dev-mode/src/using_server_process.ts#L32

Reply:
The main reason is because the dev cli implements auto restart on changes and ensuring that we don't see failed requests while it's restarting, and so that there's a single command to run that sets that and the optimizer up.

Reply:
If you instead run `node scripts/cli serve` without `--dev`, or instead of using `yarn start`, you can run the server without the dev cli wrapper that makes dev a lot easier in most cases.